src/core/src/model.cpp

```diff
@@ -298,6 +298,8 @@ std::vector<shared_ptr<ov::Node>> ov::Model::get_ordered_ops() const {
         m_cached_ordered_ops.push_back(node);
         node->insert_info(m_shared_rt_info);
     });
+    m_cached_output_names.clear();
```

Comment:
Why do you need to clear these vectors in this method?

Reply:
Yes, this is needed, but I found that test cases verifying this need are absent. Added appropriate test cases which will be failed if this 'clear' is absent. Brief scenario:
```
// Scenario:
// 1. Create model with nodes A,B,C.
// 2. Add output to some node 'A' - 'names cache' will be created
// 3. Remove some node 'B' - ordered_ops_cache will be 'false'
// 4. Assign name 'B' to existing node 'C'
// 5. Call get_ordered_ops (ordered_ops_cache=true)
// 6. Expect: Add_output to 'B' - output to node 'C' shall be added on step 4
// Without clearing 'names cache' on step 5 - test will incorrectly add output to 'B'
```